R/oc_forward.R

```diff
@@ -0,0 +1,328 @@
+#'
+#' @seealso \code{\link{oc_reverse}} for reverse geocoding, and the
+#'   \href{https://opencagedata.com/api}{OpenCage API documentation} for more
+#'   information about the API.
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' oc_forward(placename = "Sarzeau")
+#' oc_forward(placename = "Islington, London")
+#' oc_forward(placename = "Triererstr 15,
+#'                         Weimar 99423,
+#'                         Deutschland")
+#' }
+#'
+oc_forward <-
+  function(placename,
+           return = c("df_list", "json_list", "geojson_list", "url_only"),
+           key = oc_key(),
+           bounds = NULL,
+           countrycode = NULL,
+           language = NULL,
+           limit = 10L,
+           min_confidence = NULL,
+           no_annotations = TRUE,
+           no_dedupe = FALSE,
+           no_record = FALSE,
+           abbrv = FALSE,
+           add_request = FALSE,
+           ...) {
+
+    # check a placename is provided
+    if (missing(placename) || is.null(placename)) {
+      stop(call. = FALSE, "You must provide a `placename`.")
+    }
+
+    # check return
+    return <- match.arg(return)
+
+    # check arguments
+    oc_check_query(
+      placename = placename,
+      key = key,
+      bounds = bounds,
+      countrycode = countrycode,
+      language = language,
+      limit = limit,
+      min_confidence = min_confidence,
+      no_annotations = no_annotations,
+      no_dedupe = no_dedupe,
+      no_record = no_record,
+      abbrv = abbrv,
+      add_request = add_request
+    )
+    # process request
+    oc_process(
+      placename = placename,
+      return = return,
+      key = key,
+      bounds = bounds,
+      countrycode = countrycode,
+      language = language,
+      limit = limit,
+      min_confidence = min_confidence,
+      no_annotations = no_annotations,
+      no_dedupe = no_dedupe,
+      no_record = no_record,
+      abbrv = abbrv,
+      add_request = add_request
+    )
+  }
+
+#' @rdname oc_forward
+#' @param data A data frame
+#' @param bind_cols logical Bind source and results data frame?
+#' @param output A character vector of length one indicating whether only
+#'   latitude, longitude and formatted address (\code{short}) or whether all
+#'   results (\code{all}) should be returned.
+#'
+#' @details
+#' \code{oc_forward_df} also accepts unquoted column names for all arguments
+#' except \code{key}, \code{return}, and \code{no_record}.
+#'
+#' @export
+oc_forward_df <-
+  function(data,
+           placename,
+           bind_cols = TRUE,
+           output = c("short", "all"),
+           key = oc_key(),
+           bounds = NULL,
+           countrycode = NULL,
+           language = NULL,
+           limit = 1L,
+           min_confidence = NULL,
+           no_annotations = TRUE,
+           no_dedupe = FALSE,
+           no_record = FALSE,
+           abbrv = FALSE,
+           ...) {
+
+    placename <- data[[deparse(substitute(placename))]]
```

Comment:
could you add a few more comments in the code just to make the different parts clear? E.g.

# argument parsing
# language
# country

etc?

Reply:
This will just be one line per argument with tidyeval.